client.go

```diff
@@ -184,6 +187,18 @@ func (c *Client) WithDialConfig(dc amqp.Config) *Client {
 	return c
 }
 
+// WithDialTimeout sets the DialTimeout and handshake deadlines to timeout.
+func (c *Client) WithDialTimeout(timeout time.Duration) *Client {
+	c.dialconfig = amqp.Config{
+		Dial: DefaultDialer(DialConf{
+			DialTimeout: timeout,
+			Deadline:    timeout,
```

Comment:
I'm not sure about the handshake deadline. Should that always be set to the same as the dial timeout?